lib/client.js

```diff
@@ -114,7 +114,24 @@ Client.prototype._connect = function (callback) {
 
   function checkPgPass (cb) {
     return function (msg) {
-      if (self.password !== null) {
+      if (typeof self.password === 'function') {
+        self._Promise.resolve()
+          .then(self.password)
```

Comment:
I didn’t comment on this one intentionally – both because of the different error-handling needs but because using `.then` here causes it to act a bit differently. Having only one error case is nice, though, I guess, so do we mind that

- `.then` passes `undefined` to `self.password` instead of no arguments, and
- the `this` value is a promise instead of `self` in `self.password`?

If we do, the fix isn’t terrible: `.then(() => self.password())`.

Reply:
Nice catch! Yes having it be explicit via `() => self.password()` is a good idea to ensure that nothing is supplied as an arg. I was thinking down the road it make make sense to add the connection parameters or some other structure to the caller similar to how the `pgPass(...)` piece works.